src/mame/misc/astrcorp.cpp

```diff
@@ -413,64 +632,61 @@ void astrocorp_state::showhanc_map(address_map &map)
+	map(0x08e000, 0x08e001).portr("EEPROM_IN");
 	map(0x090000, 0x093fff).ram().share("nvram"); // battery
-	map(0x0a0001, 0x0a0001).w(FUNC(astrocorp_state::screen_enable_w));
+	map(0x0a0000, 0x0a0001).nopr().w(FUNC(astrocorp_state::screen_enable_w)).umask16(0x00ff);
 	map(0x0e0000, 0x0e0001).r(FUNC(astrocorp_state::unk_r));
 	map(0x0e0000, 0x0e0000).w(m_oki, FUNC(okim6295_device::write));
 }
 
-// unknown read at 0x202000, 0x20a000, 0x400000
 void astrocorp_state::skilldrp_map(address_map &map)
 {
 	map(0x000000, 0x03ffff).rom();
 	map(0x200000, 0x200fff).ram().share("spriteram");
-	map(0x202000, 0x202001).w(FUNC(astrocorp_state::draw_sprites_w));
+	map(0x202000, 0x202001).nopr().w(FUNC(astrocorp_state::draw_sprites_w));
 	map(0x204000, 0x204001).portr("INPUTS");
 	map(0x208001, 0x208001).w(FUNC(astrocorp_state::eeprom_w));
-	map(0x20a000, 0x20a001).w(FUNC(astrocorp_state::skilldrp_outputs_w));
-	map(0x20e000, 0x20e001).portr("EEPROMIN");
+	map(0x20a000, 0x20a001).nopr().w(FUNC(astrocorp_state::skilldrp_outputs_w));
+	map(0x20e000, 0x20e001).portr("EEPROM_IN");
 	map(0x380000, 0x3801ff).ram().w(m_palette, FUNC(palette_device::write16)).share("palette");
-	map(0x400001, 0x400001).w(FUNC(astrocorp_state::screen_enable_w));
+	map(0x400000, 0x400001).nopr().w(FUNC(astrocorp_state::screen_enable_w)).umask16(0x00ff);
 	map(0x500000, 0x507fff).ram().share("nvram"); // battery
 	map(0x580001, 0x580001).w(FUNC(astrocorp_state::oki_bank_w));
 	map(0x600001, 0x600001).rw(m_oki, FUNC(okim6295_device::read), FUNC(okim6295_device::write));
 }
 
-// unknown read at 0x500000
 void astrocorp_state::speeddrp_map(address_map &map)
 {
 	map(0x000000, 0x01ffff).rom();
 	map(0x280000, 0x283fff).ram().share("nvram"); // battery
 	map(0x380000, 0x380fff).ram().share("spriteram");
-	map(0x382000, 0x382001).w(FUNC(astrocorp_state::draw_sprites_w));
+	map(0x382000, 0x382001).nopr().w(FUNC(astrocorp_state::draw_sprites_w));
 	map(0x384000, 0x384001).portr("INPUTS");
 	map(0x388001, 0x388001).w(FUNC(astrocorp_state::eeprom_w));
 	map(0x38a000, 0x38a001).w(FUNC(astrocorp_state::skilldrp_outputs_w));
-	map(0x38e000, 0x38e001).portr("EEPROMIN");
+	map(0x38e000, 0x38e001).portr("EEPROM_IN");
 	map(0x480000, 0x4801ff).ram().w(m_palette, FUNC(palette_device::write16)).share("palette");
-	map(0x500001, 0x500001).w(FUNC(astrocorp_state::screen_enable_w));
+	map(0x500000, 0x500001).nopr().w(FUNC(astrocorp_state::screen_enable_w)).umask16(0x00ff);
 	map(0x580001, 0x580001).w(FUNC(astrocorp_state::oki_bank_w));
 	map(0x600001, 0x600001).rw(m_oki, FUNC(okim6295_device::read), FUNC(okim6295_device::write));
 }
 
 // PC refers to magibomb:
-// - 0x22a4 reads to $a0101a, resets if 0
-// - 0x8ba if $a10001 & 0xf != 0 it writes "SEGA" to $a14000 (wtf?)
-// - all lv 1-7 irqs writes a vector to D1 then resets program flow,
+// - 0x22a4 reads from $a0101a, resets if 0
+// - 0x8ba if $a10001 & 0xf != 0 it writes "SEGA" to $a14000 (!?)
```

Comment:
This... is part of the Sega Mega Drive TMSS authentication process:
> write a long word containing "SEGA" (all caps) to address $A14000 before accessing the VDP data port ($C00000). If the game fails to do this, the next VDP data port access will cause the 68000 to lock up — this is done by the I/O controller.
> - The sample code in the official programmer documentation has this part skipped on boards that report themselves as revision 0 (byte at $A10001 & $F); the effect of doing this write on a revision 0 board is unknown.

Reply:
Interesting. Thanks for the pointer!